src/flask/blueprints.py

```diff
@@ -365,6 +365,7 @@ def add_url_rule(
         rule: str,
         endpoint: t.Optional[str] = None,
         view_func: t.Optional[t.Callable] = None,
+        provide_automatic_options: t.Optional[bool] = None,
```

Comment:
`self.record(lambda s: s.add_url_rule(rule, endpoint, view_func, provide_automatic_options=provide_automatic_options, **options))` is also required

Reply:
Good catch! I don't know enough about blueprints to know if this is the desired behavior, but it seems right to pass it properly.

It turns out the `add_url_rule()` in the lambda is a different function than the function I had already modified. It seems there is room for improvement in the type annotations for `record()`, but that is too complicated for me to handle.

Reply:
Should anything about these changes be added to the changelog as well?

Reply:
I think the changelog is clear enough. 

The record is less concerning (typing wise) as it is perhaps solely used internally by Flask.